mwclient/listing.py

```diff
@@ -9,12 +9,18 @@ class List:
     This is a class providing lazy iteration.  This means that the
     content is loaded in chunks as long as the response hints at
     continuing content.
+
+    max_items limits the total number of items that will be yielded
+    by this iterator. api_chunk_size sets the number of items that
+    will be requested from the wiki per API call (this iterator itself
+    always yields one item at a time). limit does the same as
+    api_chunk_size for backward compatibility, but is deprecated due
+    to its misleading name.
     """
 
     def __init__(self, site, list_name, prefix,
```

Comment:
Well...within mwclient, only things in listing.py instantiate `GeneratorList` directly. Other things generally do `mwclient.listing.List.get_list(generator)(blahblah)`, and I've just checked and I believe this PR touches every place that does that.

There are a couple of subclasses of `GeneratorList` - `PageList` and `PagePropertyGenerator` - but nowhere those get instantiated seems to allow the passing through of generic kwargs.

Of course someone could instantiate or subclass `GeneratorList` directly and include `limit` as a kwarg...but what problem do you foresee if somebody does? I might be dim but I'm not immediately seeing how it would be a problem? AFAICS it would get passed through to the API without our deprecation warning being shown, but...is that an issue? If you're that deep into things you probably know or ought to know what it actually does...

Sorry if I'm misunderstanding the concern :)

Reply:
I just double checked and yeah, it looks like have covered every case internal to mwclient.

And I also don't think someone would necessarily subclass GeneratorList, but I could definitely see people doing `mwclient.listing.List.get_list` in their own code, which I believe would use the existing `limit` param directly to the api - maybe that's ok though?

Reply:
hmm, looking at it...all `get_list` really does is return `List` or `GeneratorList` (the class itself, not an instance of it). You then instantiate it yourself, obviously. `GeneratorList.__init__` calls `super(GeneratorList, self).__init__(site, list_name, prefix, *args, **kwargs)` - i.e. `List.__init__` - before it does anything else, so I think any road, you wind up running through `handle_limit`, same as the other paths. Right?

Reply:
yes, you're right! okay, in this case I think we're good 🎉

Reply:
I just hacked up a dumb test, and doing `lst = GeneratorList(mock_site, 'pages', 'p', limit=2); vals = [x for x in lst]` definitely runs throuhg `handle_limit` and triggers the deprecation warning.